### DISCUSSION TITLE
Typescript React Text does not render on Canvas

### DISCUSSION THREAD

**Discussion by wibed** [pmndrs/drei#1206] [Q&A] [answered]
i try to render given the following options
```jsx

  return (
    <div className="pt-16 h-full w-full min-h-screen">
        <Canvas >
          <Text color="black" anchorX="center" anchorY="middle" font={"https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu4mxM.woff"} >
            hello world!
          </Text>  
        </Canvas>
    </div>
  )
```

cors is configured correctly, i dont get any errors nor other indication except 
that the canvas stays blank

it works with
https://codesandbox.io/s/react-three-fiber-starter-n8iz2?file=/src/index.js:147-569
```jsx
function Thing() {
  const ref = useRef()
  useFrame(() => (ref.current.rotation.x = ref.current.rotation.y += 0.01))
  return (
    <mesh
      ref={ref}
      onClick={e => console.log('click')}
      onPointerOver={e => console.log('hover')}
      onPointerOut={e => console.log('unhover')}>
      <boxBufferGeometry attach="geometry" args={[1, 1, 1]} />
      <meshNormalMaterial attach="material" />
    </mesh>
  )
}
...
  <Canvas>
    <Thing />
  </Canvas>
...
```

**Comment by drcmda** [comment, marked as answer]
there are dozens of running examples with text, here's one https://codesandbox.io/s/html-input-fields-024uom

the sandbox you posted uses ancient packages (react-three-fiber is deprecated, use @react-three/fiber)

troika default fontsize is tiny btw, it might just be that you don't see the text.

edit:

fontSize={1} works, i see the text. but i think a default that small makes little sense, i'll set it to 1.

**Reply by wibed** [reply]
the libraries i used are
```jsx
import * as THREE from 'three'
import { Canvas, useFrame } from '@react-three/fiber'
import { Text, TrackballControls } from '@react-three/drei'
```

i changed it to use
```jsx
  const fontProps = { font: "https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu4mxM.woff", fontSize: 2.5, letterSpacing: -0.05, lineHeight: 1, 'material-toneMapped': false }
  ```
  
  but had no luck

**Reply by drcmda** [reply]
i can just take the code you posted above, place it in any box, give it fontSize={1} and i see it, it must be something else in your project, maybe something is occluding it, or you forgot to spread these props over `<Text>`, or your camera is facing the wrong direction or the frustum is too small/too big, etc.

```jsx
<Text color="black" anchorX="center" anchorY="middle" fontSize={1} font="https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu4mxM.woff" >
  hello world!
</Text>
```

**Reply by wibed** [reply]
nope there is nothing over it nor under it.
it is just a single component within an existing project

its an outlet within a router v6 route, but i doubt that matters

an errorBoundary, and a browser router component, additionally to the redux and react.strictmode, wrapping the route.
it is using webpack, but the mesh example renders fine tho

**Reply by wibed** [reply]
now thats interesting,
in chrome, safari it works, but in firefox does not.

**Reply by wibed** [reply]
opened an issue thank you

https://bugzilla.mozilla.org/show_bug.cgi?id=1808012
https://github.com/pmndrs/drei/issues/1207

**Reply by drcmda** [reply]
i see text in every browser, including firefox

https://codesandbox.io/s/gifted-kirch-hwwvko?file=/src/App.js

i don't think ff has much to do with anything tbh, the text comes from the troika library, it's an sdf based shader. unless webgl as a whole isn't functional in your browser the issue has to be something else.

**Reply by wibed** [reply]
which version do you use? i dont see any text and the settings are correctly set

**Reply by makc** [reply]
firefox 108.0.1:
<img width="946" alt="Безымянный" src="https://user-images.githubusercontent.com/242577/210114451-3775cc7f-96cb-43ea-a2ae-608c61ca7fd0.png">

**Reply by wibed** [reply]
wierd!
<img width="1681" alt="image" src="https://user-images.githubusercontent.com/69305089/210118691-eaea4412-59fb-42d6-a322-cc366e1a0ee2.png">
